evm/src/memory/segments.rs

```diff
@@ -58,6 +61,7 @@ impl Segment {
             Self::StorageTrieAddresses,
             Self::StorageTriePointers,
             Self::StorageTrieCheckpointPointers,
+            Self::RipeMd,
```

Comment:
Just as a general rule of thumb, I'd suggest we should use `KERNEL_GENERAL` by default. But if we have a scenario where function A calls function B and both use `KERNEL_GENERAL`, potentially clobbering each other, it should be fine to add a new segment to avoid clobbering.

Reply:
So do you think in the case of RipeMD, I should keep it as is, giving it its own segment, or do you think it's worth changing it to `KERNEL_GENERAL`?

Reply:
Yeah I'd slightly prefer switching to `KERNEL_GENERAL`, just to have one less segment, but it's not very important, we can always do it later.